apps/web/src/app/_components/onboarding/onboarding-form.tsx

```diff
@@ -0,0 +1,256 @@
+import { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Button } from "@cooper/ui/button";
+import { DialogTitle } from "@cooper/ui/dialog";
+import { Form, FormControl, FormField, FormItem } from "@cooper/ui/form";
+import { RadioGroup, RadioGroupItem } from "@cooper/ui/radio-group";
+
+import {
+  FormLabel,
+  FormMessage,
+} from "~/app/_components/themed/onboarding/form";
+import { Input } from "~/app/_components/themed/onboarding/input";
+import { api } from "~/trpc/react";
+import { Select } from "../themed/onboarding/select";
+import { majors, monthOptions } from "./constants";
+import { BrowseAroundPrompt } from "./post-onboarding/browse-around-prompt";
+import { CoopPrompt } from "./post-onboarding/coop-prompt";
+
+const formSchema = z.object({
+  firstName: z.string().min(1, "First name is required"),
+  lastName: z.string().min(1, "Last name is required"),
+  email: z
+    .string()
+    .min(1, { message: "Email is required" })
+    .email("This is not a valid email"),
+  major: z.string().min(1, "Major is required"),
+  minor: z.string().optional(),
+  graduationYear: z.coerce
+    .number()
+    .min(2010, "Graduation year must be 2010 or later")
+    .max(2030, "Graduation year must be 2030 or earlier"),
```

Comment:
thoughts on like, current year + 6 for graduation max

Reply:
smort